src/cclib/parser/orcaparser.py

```diff
@@ -62,6 +62,114 @@ def extract(self, inputfile, line):
         if "Program Version" in line:
             self.metadata["package_version"] = line.split()[2]
 
+        # ================================================================================
+        #                                         WARNINGS
+        #                        Please study these warnings very carefully!
+        # ================================================================================
+        #
+        # Warning: TCutStore was < 0. Adjusted to Thresh (uncritical)
+        #
+        # WARNING: your system is open-shell and RHF/RKS was chosen
+        #   ===> : WILL SWITCH to UHF/UKS
+        #
+        #
+        # INFO   : the flag for use of LIBINT has been found!
+        #
+        # ================================================================================
+        if line.strip() == "WARNINGS":
+            next(inputfile)
+            self.skip_lines(inputfile, ['equals', 'blank'])
+
+            if 'warnings' not in self.metadata:
+                self.metadata['warnings'] = []
+            if 'info' not in self.metadata:
+                self.metadata['info'] = []
+            line = next(inputfile)
+            while line[0] != '=':
+                if line.lower()[:7] == 'warning':
+                    self.metadata['warnings'].append('')
+                    while len(line) > 1:
+                        self.metadata['warnings'][-1] += line[9:]
+                        line = next(inputfile)
+                elif line.lower()[:4] == 'info':
+                    self.metadata['info'].append('')
+                    while len(line) > 1:
+                        self.metadata['info'][-1] += line[9:]
+                        line = next(inputfile)
+                line = next(inputfile)
+
+        # ================================================================================
+        #                                        INPUT FILE
+        # ================================================================================
+        # NAME = input.dat
+        # |  1> %pal nprocs 4 end
+        # |  2> ! B3LYP def2-svp
+        # |  3> ! Grid4
+        # |  4>
+        # |  5> *xyz 0 3
+        # |  6>     O   0   0   0
+        # |  7>     O   0   0   1.5
+        # |  8> *
+        # |  9>
+        # | 10>                          ****END OF INPUT****
+        # ================================================================================
+        if "INPUT FILE" in line:
+            next(inputfile)
+            name = next(inputfile).split()[-1]
+            methods = []
+            line = next(inputfile)
+            coords = []
+            while line[0] == '|':
+                line = line[6:]
+                # Keywords block
+                if line[0] == '!':
+                    methods += line[1:].split()
```

Comment:
I think we currently have `metadata['methods']` as just quantum mechanical methods and not all options, where the length is related to how many calculations have been run (like if you do `$new_job`, or `@@@` in Q-Chem or `--` in Gaussian).

Reply:
Yes, let's stick to actual methods, and not properties.

Reply:
The simple keyword input in ORCA (lines with `!`) freely mixes methods and properties, thus I'm not sure how to separate the two without checking every keyword against a set of valid keywords. This seems to be what the Gaussian metadata parser does. However, when doing DFT in ORCA, the functional name is used directly, without any flag indicating DFT. How should I go about parsing this?